xls/modules/zstd/zstd_dec_test.cc

```diff
@@ -0,0 +1,241 @@
+// Copyright 2020 The XLS Authors
```

Comment:
this seems to fail w/ a timeout when introducing a deliberate failure (i.e: changing `MAGIC_NUMBER` from `u32:0xFD2FB528;` to `u32:0xFD2FB527;`
```
Target //xls/modules/zstd:zstd_dec_cc_test up-to-date:
  bazel-bin/xls/modules/zstd/zstd_dec_cc_test
INFO: Elapsed time: 300.804s, Critical Path: 300.53s
INFO: 3 processes: 3 linux-sandbox.
INFO: Build completed, 1 test FAILED, 3 total actions
//xls/modules/zstd:zstd_dec_cc_test                                     TIMEOUT in 300.0s
  /usr/local/google/home/proppy/.cache/bazel/_bazel_proppy/fb962cb496438c85ace9ac1a1a0be573/execroot/com_google_xls/bazel-out/k8-fastbuild/testlogs/xls/modules/zstd/zstd_dec_cc_test/test.log

Executed 1 out of 1 test: 1 fails locally.
```
is that expected? or is there a way to make it fail earlier?
(attached the [test.log](https://github.com/google/xls/files/14597453/test.log))

Reply:
We would imagine that in the case of passing invalid ZSTD frame (e.g. wrong `MAGIC_NUMBER`), decoder should spin in `DECODE_MAGIC_NUMBER` state trying to detect it in incoming data packets and discarding the oldest byte in the buffer each time it failed. In your case decoder got stuck in `ERROR` state which requires restarting the decoder and we will be fixing that.

When it comes to failing early we'll have to look closely if this is possible.

Reply:
Resolving, since we have other failing test that show the CI failure mode now.